Rust IPFS Milestone Report, Phase 1.2 (#44)
ipfs/devgrants: open-grants/ipfs-rust/reports/phase-1.2.md

Relevant pull requests:
- [`/bitswap`](https://github.com/ipfs-rust/rust-ipfs/pull/131)

## Miscellaneous

Relevant pull requests:
- [Subscription deadlocks](https://github.com/ipfs-rust/rust-ipfs/pull/130) were discovered and fixed
- [Flaky tests](https://github.com/ipfs-rust/rust-ipfs/pull/133) were fixed
- The [domain](https://github.com/ipfs-rust/rust-ipfs/pull/151) crate is slightly outdated, so it was pinned to a specific commit
- Small refactor to [remove some `&mut`s](https://github.com/ipfs-rust/rust-ipfs/pull/155) from the codebase
- [`clippy --workspace --tests --examples`](https://github.com/ipfs-rust/rust-ipfs/pull/156) was added to CI, and the resultant [clippy warnings](https://github.com/ipfs-rust/rust-ipfs/pull/159) were dealt with

## Interface Test Refactoring

> It was discovered during this effort that many if the interface and interop tests use a diversity of APIs, not just the ones being tested, i.e. `ipfs.add` is used in the `refs.local` tests. With the philosophy that the tests should utilize "lower-level" APIs such as `ipfs.block.put`.

Reviewer: ```suggestion
It was discovered during this effort that many of the interface and interop tests use a diversity of APIs, not just the ones being tested, i.e. `ipfs.add` is used in the `refs.local` tests. With the philosophy that the tests should utilize "lower-level" APIs such as `ipfs.block.put`.
```

> With permission from PL, a number of PRs were made to js-ipfs, two of which have already been merged:

Reviewer: ```suggestion
As part of this work, a number of PRs were made to js-ipfs, two of which have already been merged:
```
this is up to the lead js-ipfs maintainers - not PL as a company.
